client/client.go

```diff
@@ -106,22 +106,32 @@ func getProjects(serviceAccountKeyJSON []byte, logger hclog.Logger, filter strin
 	}
 
 	projects := make([]string, 0)
+	inactiveProjects := 0
 	for {
 		output, err := call.Do()
 		if err != nil {
-			return projects, err
+			return nil, err
 		}
 		for _, project := range output.Projects {
 			if project.LifecycleState == "ACTIVE" {
 				projects = append(projects, project.ProjectId)
 			} else {
 				logger.Info("Project state is not active. Project will be ignored", "project_id", project.ProjectId)
+				inactiveProjects++
 			}
 		}
 		if output.NextPageToken == "" {
 			break
 		}
 		call.PageToken(output.NextPageToken)
 	}
+
+	if len(projects) == 0 {
+		if inactiveProjects > 0 {
+			return nil, fmt.Errorf("project listing failed: no active projects")
+		}
+		return nil, fmt.Errorf("project listing failed")
```

Comment:
the listing didn't fail, but we have no projects to iterate on, so the error should be different message IMO.

Reply:
not a GCP expert, and the reason for that PR is I'm unable to set permissions correctly for my readonly user so it can't list projects. Feel free to suggest a message.